src/shogun/lib/SGVector.cpp

```diff
@@ -157,19 +169,22 @@ void SGVector<T>::set_const(T const_elem)
 template<>
 void SGVector<float64_t>::set_const(float64_t const_elem)
 {
+	on_gpu_check();
 	catlas_dset(vlen, const_elem, vector, 1);
 }
 
 template<>
 void SGVector<float32_t>::set_const(float32_t const_elem)
 {
+	on_gpu_check();
 	catlas_sset(vlen, const_elem, vector, 1);
 }
 #endif // HAVE_CATLAS
 
 template<class T>
 void SGVector<T>::range_fill(T start)
 {
+	on_gpu_check();
```

Comment:
for your later TODO list
We want to remove all the methods that you put this check in here.
And then we want to make them part of linalg